pkgs/development/python-modules/devito/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchFromGitHub
+, anytree
+, pytest-cov
+, codecov
+, nbval
+, flake8
+, numpy
+, sympy
+, scipy
+, cached-property
+, psutil
+, py-cpuinfo
+, cgen
+, click
+, multidict
+, distributed
+, pytest-runner
+, pyrevolve
+, codepy
+}:
+
+buildPythonPackage rec {
+  pname = "devito";
+  # grabbing the commit that has NixOS compatibility fix
+  version = "4.6.2-bbad3b1";
+
+  src = fetchFromGitHub {
+    repo = "devito";
+    owner = "devitocodes";
+    rev = "bbad3b17be8c9061ca1967760827b2e4b1f11afe";
+    sha256 = "1yd89jh897fkc7089smgnxmgkvwkd0bsap1i93ixmh7pwzjsbdm3";
+  };
+
+  # several tests fail but they pass outside the build (I've tested it); they are also very slow.
```

Comment:
How? Maybe we can easily fix them?

Reply:
Hm, most of them are MPI related, which I assume don't work because of network issues. Also troubleshooting them is extremely stressful bcs it takes hours to complete them all (this is a HPC library after all).
I can run it and put the log in a pastebin either way, I sincerly wish we could fix the testing phase so that I could offload that to hydra.

Reply:
Sounds more like LPC to me 😉 

If we can't figure them out thats not great but must work for now. Maybe pytest-xdist is needed?

Reply:
Haha,

hm, I dunno about the pytest-xdist. I'm finishing up all your other requested changes and then I will let the tests run, giving you the pastebin result in the end.

Reply:
Finally managed to do it!
[](https://pastebin.com/qtuEuqzF)